app/models/ems_refresh/refreshers/ems_refresher_mixin.rb

```diff
@@ -103,7 +103,24 @@ def collect_inventory_for_targets(ems, targets)
         # override this method and return an array of:
         #   [[target1, inventory_for_target1], [target2, inventory_for_target2]]
 
-        collect_legacy_inventory_for_targets(ems)
+        provider_module = ManageIQ::Providers::Inflector.provider_module(ems.class).name
+
+        targets_to_collectors = targets.each_with_object({}) do |target, memo|
+          # expect collector at <provider>/Inventory/Collectors/<target_name>
+          memo[target] = "#{provider_module}::Inventory::Collectors::#{target.class.name.demodulize}".safe_constantize
```

Comment:
I wonder how this will work for

ManageIQ::Providers::Amazon::Inventory::Collector::StorageManager::Ebs